### PR TITLE
Stop lazy migration for guests on hypervisors

### PR DESCRIPTION
**Pull request by wottop**
The migration will be handled on the hypervisors' guests after the check in instead of on next entitlement listing. The entitlement check was originally moved because the process was synchronous for hypervisor reporting and that caused timeouts. Now that it is an asynchronous process, we will move the check back to each hypervisor report. This way the overall system metrics can be  accurate.

### CODE REVIEW

**Comment by vritant** [comment]
cause for spec test failure is fixed in #1164, yet to be merged. Apologies for the inconvenience.

**Comment by cnsnyder** [comment]
UpdateConsumer is not called for guest consumers on async hypervisor check in. The logic in UpdateConsumer should likely go in CheckForGuestsUpdate

**Comment by wottop** [comment]
CheckForGuestsUpdate is before the hypervisor is stored. I added the code to the async process.

**Comment by cnsnyder** [comment]
There is one case that doesn't seem to work for me here. If I've reported that a guest exists on two different hypervisors (lets call them A and B and assume we receive the reports in order A, B). Then if we later receive a report indicating that the guest is no longer on hypervisor B. If we subsequently receive a report indicating that the guest is still on hypervisor A (and no other changes have occurred to hypervisor A) then we end up not checking for migration of the guest back to hypervisor A.

**Comment by cnsnyder** [comment]
We need to check for migration when the guests are added in the CheckForGuestsUpdate method.
We will already be iterating over the list of existing/added guest ids.

**Comment by wottop** [comment]
buildbot: retest this please

File: server/spec/hypervisor_check_in_spec.rb [hunk 1]
```diff
@@ -619,4 +619,122 @@ def create_virtwho_client(user)
...
   end
+
+it 'should allow a single guest to be migrated and revoke host-limited ents' do
+    owner = create_owner random_string('test_owner1')
+    user = user_client(owner, random_string("user"))
+    virtwho = create_virtwho_client(user)
+    uuid1 = random_string('system.uuid')
+    guests = [{:guestId => uuid1}]
+
+    host_consumer = user.register(random_string('host1'), :hypervisor, nil,
```
Comment (cnsnyder) [inline server/spec/hypervisor_check_in_spec.rb]: Any reason this consumer is a "hypervisor" while the one just below it is a "system".

File: server/spec/hypervisor_check_in_spec.rb [hunk 2]
```diff
@@ -619,4 +619,122 @@ def create_virtwho_client(user)
...
+    new_consumer_client = Candlepin.new(nil, nil, new_host_consumer['idCert']['cert'], new_host_consumer['idCert']['key'])
+    guest_client = Candlepin.new(nil, nil, guest_consumer['idCert']['cert'], guest_consumer['idCert']['key'])
+
+    virtwho.hypervisor_check_in(owner['key'], get_host_guest_mapping('hypervisor_id_1', [uuid1]))
+
+    # consumer_client.update_guestids(guests)
+    pools = consumer_client.list_pools :consumer => host_consumer['uuid']
+    pools.length.should == 1
+    consumer_client.consume_pool(pools[0]['id'], {:quantity => 1})
+    new_consumer_client.consume_pool(pools[0]['id'], {:quantity => 1})
```
Comment (cnsnyder) [inline server/spec/hypervisor_check_in_spec.rb]: Should new_consumer_client consume from the same pool that consumer_client does?

**Comment by cnsnyder** [comment]
Looks fine to me. Will merge after tests pass.

**Comment by cnsnyder** [comment]
buildbot: retest this please

**Comment by cnsnyder** [comment]
@wottop Do not pull. An amend went sideways. Please repush your local branch to this one.

**Comment by cnsnyder** [comment]
@wottop Thanks!

**Pull request merged**